extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java

```diff
@@ -0,0 +1,188 @@
+package io.quarkus.amazon.lambda.deployment;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import org.apache.commons.compress.archivers.zip.ZipArchiveEntry;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream;
+import org.apache.commons.compress.utils.IOUtils;
+import org.jboss.logging.Logger;
+
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.pkg.builditem.ArtifactResultBuildItem;
+import io.quarkus.deployment.pkg.builditem.JarBuildItem;
+import io.quarkus.deployment.pkg.builditem.NativeImageBuildItem;
+import io.quarkus.deployment.pkg.builditem.OutputTargetBuildItem;
+import io.quarkus.deployment.pkg.steps.NativeBuild;
+
+/**
+ * Generate deployoment package zip for lambda.
+ *
+ */
+public class FunctionZipProcessor {
+    private static final Logger log = Logger.getLogger(FunctionZipProcessor.class);
+
+    /**
+     * Function.zip is same as the jar plus dependencies in lib/ if not uberjar
+     * plus anything in src/main/zip.jvm
+     *
+     * @param target
+     * @param artifactResultProducer
+     * @param jar
+     * @throws Exception
+     */
+    @BuildStep(onlyIf = IsNormal.class, onlyIfNot = NativeBuild.class)
+    public void jvmZip(OutputTargetBuildItem target,
+            BuildProducer<ArtifactResultBuildItem> artifactResultProducer,
+            JarBuildItem jar) throws Exception {
+        log.info("**jvmZip");
+        Path zipPath = target.getOutputDirectory().resolve("function.zip");
```

Comment:
logging in this method looks like it was used for debug

Reply:
yikes, lol!